staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/types.go

```diff
@@ -53,13 +53,23 @@ type APIServiceSpec struct {
 	// CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
 	CABundle []byte
 
-	// Priority controls the ordering of this API group in the overall discovery document that gets served.
-	// Client tools like `kubectl` use this ordering to derive preference, so this ordering mechanism is important.
-	// Values must be between 1 and 1000
-	// The primary sort is based on priority, ordered lowest number to highest (10 before 20).
+	// GroupPriority controls the ordering of the group with respect to other groups. The lowest of all GroupPriorities
+	// for the same group is used.  Must be greater than zero.
+	// The primary sort is based on GroupPriority, ordered lowest number to highest (10 before 20).
+	// The secondary sort is based on the VersionPriority, ordered lowest number to highest.
+	// The tertiary sort is based on the alphabetical comparison of the name of the object.  (v1.bar before v1.foo)
+	// We'd recommend something like: *.k8s.io (except extensions) at 1000, extensions at 2000
+	// PaaSes (OpenShift, Deis) are recommended to be in the 3000s
+	GroupPriority int32
```

Comment:
> David argued that that folks might want different versions served from different places in case of a migration or something, a use case I'm not completely convinced is actually legit

Letting you register a service representing v2 of an API, backed by a pool of v2 servers, while keeping the v1 service as-is, is hard to argue against

> It's not like it's impossible to work around the existing Priority field to get things to show up in the order you want (use the first 2 digits for the group and the last two for the version, like I suggested).

Given that this is the means to register external APIs to register, merging two intents into a single int field by convention seems like a really bad idea.